### ISSUE TITLE
mpaboxStatic coordinates: 'auto' requires zoom

### ISSUE THREAD

**Issue by chrbradley** [mapbox/mapbox-sdk-js#268]
I'm trying to use mapboxStatic with `coordinates: 'auto'`. When I omit `zoom:` it returns an error, which is unexpected because the intent of setting `coordinates: 'auto'` is for the service to figure out the zoom level for me:

```
      .getStaticImage({
        ownerId: 'mapbox',
        styleId: 'streets-v10',
        width: 300,
        height: 300,
        coordinates: 'auto',
        // zoom: 2,
        overlays: [
          {
            marker: {
              coordinates: [-73.9249, 40.6943],
            },
          },
          {
            marker: {
              coordinates: [-87.6861, 41.8373],
            },
          },
        ],
      });
```
and the error:
`client err Error: zoom is required.`

Including `zoom: 2` results in the following url which doesn't resolve to a valid path and returns an error.

url:
`https://api.mapbox.com/styles/v1/mapbox/streets-v10/static/pin-s(-73.9249,40.6943),pin-s(-87.6861,41.8373)/auto,2/300x300`

error:
`{"message":"Longitude, latitude, zoom, bearing and pitch must all be numbers."}`

When I remove the zoom value from the url, `.../auto,2/...` -> `.../auto/...` the url resolves to a valid map.

**Comment by davidtheclark** [comment]
Thanks again, @chrbradley. Looks like we need to tweak two things: the validation of the parameters in mapbox-sdk-js, and the documentation in https://www.mapbox.com/api-documentation/#retrieve-a-static-map-from-a-style that suggests `zoom` is always required. I'll have fixes up soon.

**Comment by davidtheclark** [comment]
@chrbradley Thanks again for the helpful issues. Both fixes should be released now in v0.4.0.
